### PR TITLE
Add filter implementations

### PR DESCRIPTION
**Pull request by aljones15**
1. Adds new function `filterImplementations`
2. Adds new function `filterByTag`
3. Improves README.md with tags explanation
4. Add `data` and `statusCode` to return
5. Warn if an oauth2 token is not set, but let the request happen.

### CODE REVIEW

File: lib/main.js [hunk 1]
```diff
@@ -7,3 +7,57 @@ import {Implementation} from './Implementation.js';
...
+ * filter the map's entries on that returns true or false.
+ *
+ * @returns {object.<string, Map>} Returns an object with matching
+ *   and non-matching maps.
+ */
+export const filterImplementations = ({map = implementations, predicate}) => {
+  const match = new Map();
+  const nonMatch = new Map();
+  for(const [key, value] of map) {
+    const result = predicate({key, value});
```
Comment (msporny) [inline lib/main.js:30]:
```suggestion
    const result = filter({key, value});
```

File: lib/main.js [hunk 2]
```diff
@@ -7,3 +7,57 @@ import {Implementation} from './Implementation.js';
...
+    if(result === true) {
+      match.set(key, value);
+    } else {
+      nonMatch.set(key, value);
+    }
```
Comment (msporny) [inline lib/main.js:35]: Seems like you might want to use Javascript's built-in Array map/reduce functionality here?

File: implementations/Mavennet.json [hunk 3]
```diff
@@ -10,11 +10,11 @@
   "issuers": [{
     "id": "did:key:z6MkiTsvjrrPNDZ1rrg9QDEYCFWCmEswT6U2cEkScb7edQ9b",
     "endpoint": "https://api.neo-flow.com/credentials/issueCredential",
-    "tags": ["VC-HTTP-API", "OAUTH2"]
+    "tags": ["OAUTH2"]
```
Comment (msporny) [inline implementations/Mavennet.json:13]:
```suggestion
    "tags": ["OAuth2"]
```

We'll want to use proper capitalization in the tags.

File: README.md [hunk 4]
```diff
@@ -87,6 +87,15 @@ Implementations are json in the following form:
...
 
+### Tags
+Tags tell the test suites which tests your issuer, verifiers, and status lists should be run on.
+
+VC-API - This tag will run [this test on your issuer](https://github.com/w3c-ccg/vc-api-issuer-test-suite) and [this test on your verifier.](https://github.com/w3c-ccg/vc-api-verifier-test-suite)
+
+Ed25519Signature2020 - This tag will run the [Ed25519 tests](https://github.com/w3c-ccg/di-ed25519-test-suite) on either your issuer and/or verifier.
+
+StatusList2021 - This will run the [Status List 2021 tests](https://github.com/w3c-ccg/status-list-2021-test-suite) on your issuer and/or verifier.
+
```
Comment (msporny) [inline README.md:98]: Is this missing `OAuth2`, and `ZCAP` tags?

Comment (aljones15) [reply README.md:98]: the idea behind the `oauth2` tag and `zcap` tag is to make it easier to test if our `oauth2` code works with other implementations in the cohort. So can do something like `filterByTag({issuerTags: ['oath2']})`. So didn't think it was worth adding here.

File: implementations/Mavennet.json (hunk 3 above)
Comment (aljones15) [reply implementations/Mavennet.json:13]: All of the other tags are SHOUTY-KEBAB-CASE so this seems out of character.

File: lib/main.js (hunk 2 above)
Comment (aljones15) [reply lib/main.js:35]: `match` and `nonMatch` are `Map`s which don't currently have `map/reduce` implemented for them yet. Although we do something like `new Map(foo.reduce(bar => [key, value]))`.

File: lib/main.js [hunk 5]
```diff
@@ -7,3 +7,57 @@ import {Implementation} from './Implementation.js';
...
+ *
+ * @param {object} options - Options to use.
+ * @param {Map} options.map - A Map.
+ * @param {Function<boolean>} options.predicate - A function to
+ * filter the map's entries on that returns true or false.
+ *
+ * @returns {object.<string, Map>} Returns an object with matching
+ *   and non-matching maps.
+ */
+export const filterImplementations = ({map = implementations, predicate}) => {
```
[the remark this answers is not in the document]
Comment (aljones15) [reply lib/main.js:26]: good idea here.

File: lib/main.js (hunk 1 above)
Comment (aljones15) [reply lib/main.js:30]: this is a good idea.

File: README.md [hunk 6]
```diff
@@ -87,6 +87,15 @@ Implementations are json in the following form:
 Please note: implementations may have security using oauth2 or zcaps, but not both.
 Implementations may also contain no security (do not add a OAUTH2 or ZCAP section in that case).
 
+### Tags
+Tags tell the test suites which tests your issuer, verifiers, and status lists should be run on.
+
+VC-API - This tag will run [this test on your issuer](https://github.com/w3c-ccg/vc-api-issuer-test-suite) and [this test on your verifier.](https://github.com/w3c-ccg/vc-api-verifier-test-suite)
```
Comment (JSAssassin) [inline README.md:93]:
```suggestion
VC-API - This tag will run the [vc-api-issuer tests](https://github.com/w3c-ccg/vc-api-issuer-test-suite) on your issuer and the [vc-api-verifier tests](https://github.com/w3c-ccg/vc-api-verifier-test-suite) on your verifier.
```

File: README.md [hunk 7]
```diff
@@ -87,6 +87,15 @@ Implementations are json in the following form:
...
+
+Ed25519Signature2020 - This tag will run the [Ed25519 tests](https://github.com/w3c-ccg/di-ed25519-test-suite) on either your issuer and/or verifier.
+
+StatusList2021 - This will run the [Status List 2021 tests](https://github.com/w3c-ccg/status-list-2021-test-suite) on your issuer and/or verifier.
```
Comment (JSAssassin) [inline README.md:97]:
```suggestion
StatusList2021 - This tag will run the [Status List 2021 tests](https://github.com/w3c-ccg/status-list-2021-test-suite) on your issuer and/or verifier.
```

**Pull request merged**